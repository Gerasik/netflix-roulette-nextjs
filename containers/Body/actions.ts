import ActionType from './constants';
import { ActionCreator } from './models';

export const setData: ActionCreator.SetData = payload => ({
  type: ActionType.SET_DATA,
  payload,
});

export const setStartData: ActionCreator.SetStartData = () => ({
  type: ActionType.SET_START_DATA,
});

// export const increment: Action. = payload => ({
//   type: ActionType.,
//   payload,
// });
