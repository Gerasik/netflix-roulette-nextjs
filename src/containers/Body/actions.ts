import ActionType from './constants';
import { ActionCreator } from './models';

export const setData: ActionCreator.SetData = payload => ({
  type: ActionType.SET_DATA,
  payload,
});

export const setStartData: ActionCreator.SetStartData = () => ({
  type: ActionType.SET_START_DATA,
});

export const addData: ActionCreator.AddData = () => ({
  type: ActionType.ADD_DATA,
});
