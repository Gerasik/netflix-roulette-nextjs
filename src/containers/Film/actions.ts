import ActionType from './constants';
import { ActionCreator } from './models';

export const setFilmData: ActionCreator.SetFilmData = payload => ({
  type: ActionType.SET_FILM_DATA,
  payload,
});

export const setFilmId: ActionCreator.SetFilmId = payload => ({
  type: ActionType.SET_FILM_ID,
  payload,
});
