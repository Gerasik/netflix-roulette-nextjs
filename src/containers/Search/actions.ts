import ActionType from './constants';
import { ActionCreator } from './models';

export const changeSearchString: ActionCreator.ChangeSearchString = payload => ({
  type: ActionType.CHANGE_SEARCH_STRING,
  payload,
});

export const changeSearchBy: ActionCreator.ChangeSearchBy = payload => ({
  type: ActionType.CHANGE_SEARCH_BY,
  payload,
});

export const changeSortBy: ActionCreator.ChangeSortBy = payload => ({
  type: ActionType.CHANGE_SORT_BY,
  payload,
});
