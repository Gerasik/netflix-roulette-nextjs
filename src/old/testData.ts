/* eslint-disable @typescript-eslint/camelcase */
import { fromJS } from 'immutable';

import { Movie, MoviesResponse, MoviesData, MovieMap } from '../../models';
import { State } from '../../containers/Body/models';

export const testMovie1: Movie | MovieMap = {
  id: 1,
  title: 'first movie',
  tagline: 'test',
  vote_average: 55,
  vote_count: 66,
  release_date: '2020-05-01',
  poster_path: '/movie_url/image.jpg',
  overview: 'ove',
  budget: 54545,
  revenue: 5454,
  runtime: 5454,
  genres: ['drama', 'sitcom', 'drama'],
};

export const testMovie2: Movie | MovieMap = {
  id: 2,
  title: 'second movie',
  tagline: 'test',
  vote_average: 55,
  vote_count: 66,
  release_date: '2020-05-01',
  poster_path: '/movie_url/image2.jpg',
  overview: 'ove',
  budget: 54545,
  revenue: 5454,
  runtime: 5454,
  genres: ['drama'],
};

export const testMoviesData: MoviesData = fromJS([testMovie1, testMovie2]);

export const testMovieResponse: MoviesResponse = fromJS({
  data: testMoviesData,
  total: 5,
  offset: 5,
  limit: 5,
});

export const testBodyReducerData = {
  moviesResponse: {},
};
export const testSearchReducerData = {
  searchString: '',
  searchBy: 'genres',
  sortBy: 'vote_average',
};

export const testInitialState: State = fromJS({
  bodyReducer: testBodyReducerData,
  searchReducer: testSearchReducerData,
});

export const testEmptyInitialState: State = fromJS({
  bodyReducer: {
    data: [],
    total: 0,
    offset: 5,
    limit: 5,
  },
  searchReducer: testSearchReducerData,
});

export const testNextState: State = fromJS({
  moviesResponse: testMovieResponse,
});
