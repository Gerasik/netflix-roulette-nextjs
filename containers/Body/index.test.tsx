/* eslint-disable @typescript-eslint/camelcase */
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

import { testMovieResponse, testNextState, testInitialState } from 'testData';
import bodyReducer from './reducer';
import BodyContainer from '.';
import { setData } from './actions';
import ActionType from './constants';

const middleware = [];
const mockStore = configureStore(middleware);

describe('Test dispatch action', () => {
  const store = mockStore(testInitialState);

  it('action `setData`', () => {
    store.dispatch(setData(testMovieResponse));

    const actions = store.getActions();
    const expectedPayload = { type: ActionType.SET_DATA, payload: testMovieResponse };

    expect(actions).toEqual([expectedPayload]);
  });
});

describe('Test initial BodyContainer', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(testInitialState);
    wrapper = mount(
      <Provider store={store}>
        <BodyContainer />
      </Provider>
    );
  });

  it('should render with initial state data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with next state data', () => {
    store.dispatch(setData(testMovieResponse));
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Test reducer', () => {
  // it('should return the initial state', () => {
  //   expect(bodyReducer(undefined, { type: 'unknown', payload: undefined })).toEqual(
  //     testInitialState
  //   );
  // });

  it('should return the state with movie response', () => {
    expect(bodyReducer(undefined, setData(testMovieResponse))).toEqual(testNextState);
  });
});
