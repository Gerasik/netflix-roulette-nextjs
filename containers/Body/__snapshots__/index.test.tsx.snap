// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test initial BodyContainer should render with initial state data 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Body)>
    <Body
      changeSortBy={[Function]}
      moviesData={
        Object {
          "data": Immutable.List [],
          "sortBy": "vote_average",
          "total": 0,
        }
      }
      setStartData={[Function]}
    >
      <div
        className="movie-container"
      >
        <header
          className="movie-header"
        >
          <span
            className="movie-count"
          />
          <div
            className="movie-buttons"
          >
            <span>
              sort by
            </span>
            <button
              className="sort-release"
              onClick={[Function]}
              type="button"
            >
              release date
            </button>
            <button
              className="sort-rating active"
              onClick={[Function]}
              type="button"
            >
              rating
            </button>
          </div>
        </header>
        <div
          className="movie-empty-result"
        >
          <span>
            No films found
          </span>
        </div>
      </div>
    </Body>
  </Connect(Body)>
</Provider>
`;

exports[`Test initial BodyContainer should render with next state data 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Body)>
    <Body
      changeSortBy={[Function]}
      moviesData={
        Object {
          "data": Immutable.List [],
          "sortBy": "vote_average",
          "total": 0,
        }
      }
      setStartData={[Function]}
    >
      <div
        className="movie-container"
      >
        <header
          className="movie-header"
        >
          <span
            className="movie-count"
          />
          <div
            className="movie-buttons"
          >
            <span>
              sort by
            </span>
            <button
              className="sort-release"
              onClick={[Function]}
              type="button"
            >
              release date
            </button>
            <button
              className="sort-rating active"
              onClick={[Function]}
              type="button"
            >
              rating
            </button>
          </div>
        </header>
        <div
          className="movie-empty-result"
        >
          <span>
            No films found
          </span>
        </div>
      </div>
    </Body>
  </Connect(Body)>
</Provider>
`;
